packages/experimental/NativeDatePicker/src/NativeDatePicker.tsx

```diff
@@ -0,0 +1,83 @@
+import { NativeModules } from 'react-native';
+export const NativeDatePicker = NativeModules.MSFDatePickerManager;
+export const { MSFDateTimePickerMode, MSFDateTimePickerDatePickerType, MSFDateTimePickerDateRangePresentation } = NativeDatePicker.getConstants();
+
+type NativeDatePickerMode = keyof typeof MSFDateTimePickerMode;
+type NativeDatePickerType = keyof typeof MSFDateTimePickerDatePickerType;
+type NativeDatePickerDateRangePresentation = keyof typeof MSFDateTimePickerDateRangePresentation;
+
+interface DatePickerParameterObject {
+    mode?: NativeDatePickerMode;
+    dateRangePresentation?: NativeDatePickerDateRangePresentation;
+    datePickerType?: NativeDatePickerType;
+    startDate?: Date;
+    endDate?: Date;
+    startTitle?: string;
+    startSubtitle?: string;
+    startTab?: string;
+    endTitle?: string;
+    endSubtitle?: string;
+    endTab?: string;
+    dateTitle?: string;
+    dateSubtitle?: string;
+    timeTitle?: string;
+    timeSubtitle?: string;
+    callback: Function;
+}
+
+NativeDatePicker.present = ({
+    mode = "date",
+    dateRangePresentation = "tabbed",
+    datePickerType = "calendar",
+    startDate = null,
+    endDate = null,
+    startTitle = null,
+    startSubtitle = null,
+    startTab = null,
+    endTitle = null,
+    endSubtitle = null,
+    endTab = null,
+    dateTitle = null,
+    dateSubtitle = null,
+    timeTitle = null,
+    timeSubtitle = null,
+    callback,
+}: DatePickerParameterObject) => {
+    NativeDatePicker.presentWithMode(
+        mode,
+        dateRangePresentation,
+        datePickerType,
+        startDate?.toISOString(),
+        endDate?.toISOString(),
+        startTitle,
+        startSubtitle,
+        startTab,
+        endTitle,
+        endSubtitle,
+        endTab,
+        dateTitle,
+        dateSubtitle,
+        timeTitle,
+        timeSubtitle,
+        callback,
+    );
+}
+
+// We get date values back from the native side as strings in ISO 8601 format and UTC.
+// We want to immediately put them back into `Date` objects in local time.
+NativeDatePicker.parseISOString = (dateISOString: string): Date => {
+    if (dateISOString == null) {
+        return null
+    }
+    var dateParts: number[] = dateISOString.split(/\D+/).map(x => parseInt(x, 10));
+    dateParts[1]--; // Date.UTC's `month` arg is zero-based
+    var dateUTC = Date.UTC.apply(null, dateParts);
+    return new Date(dateUTC);
```

Comment:
We can't do something like Date.parse()? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse

Reply:
I decided to write an explicit parser because of this note in [Date() constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date):

> Note: Parsing of date strings with the Date constructor (and Date.parse(), which works the same way) is strongly discouraged due to browser differences and inconsistencies.

Reply:
I wonder if it's more standardized across React Native, but even then there's multiple javascript engines that could be used. I tend to be against manual date parsing (as I mentioned in a different comment) but I suppose this is the way to ensure it's always correct? Fingers crossed :D